bpf/lib/nodeport.h

```diff
@@ -1057,14 +1096,10 @@ static __always_inline int nodeport_lb4(struct __ctx_buff *ctx,
 		if (svc)
 			return DROP_IS_CLUSTER_IP;
 
-		if (nodeport_uses_dsr4(&tuple)) {
-			return CTX_ACT_OK;
-		} else {
-			ctx_store_meta(ctx, CB_NAT, NAT_DIR_INGRESS);
-			ctx_store_meta(ctx, CB_SRC_IDENTITY, src_identity);
-			ep_tail_call(ctx, CILIUM_CALL_IPV4_NODEPORT_NAT);
-			return DROP_MISSED_TAIL_CALL;
-		}
+		ctx_store_meta(ctx, CB_NAT, NAT_DIR_INGRESS);
+		ctx_store_meta(ctx, CB_SRC_IDENTITY, src_identity);
+		ep_tail_call(ctx, CILIUM_CALL_IPV4_NODEPORT_NAT);
+		return DROP_MISSED_TAIL_CALL;
```

Comment:
Do we also need this if ip masq is disabled and we run with full DSR?

Reply:
Good spot, will add #ifdef.